libs/pika/async_mpi/CMakeLists.txt

```diff
@@ -11,6 +11,33 @@ if(NOT ${PIKA_WITH_MPI})
   return()
 endif()
 
+# For testing and benchmarking, we would like to know if mpi continuations extension is present
+try_run(
+  # Name of variable to store the run result (process exit status; number) in:
+  MPIX_RUN_RESULT
+  # Name of variable to store the compile result (TRUE or FALSE) in:
+  MPIX_COMPILE_RESULT
+  # Binary directory:
+  ${CMAKE_CURRENT_BINARY_DIR}/
+  # Source file to be compiled:
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/check_openmpi_continuations.cpp
+  # Where to store the output produced during compilation:
+  COMPILE_OUTPUT_VARIABLE TEST_COMPILE_OUTPUT
+  # Where to store the output produced by running the compiled executable:
+  RUN_OUTPUT_VARIABLE TEST_RUN_OUTPUT
+)
+
+# Did compilation succeed and process return 0 (success)?
+if("${MPIX_COMPILE_RESULT}" AND ("${MPIX_RUN_RESULT}" EQUAL 0))
+  message(STATUS "MPIx Continuations detected")
+  set(MPIX_CONTINUATIONS_AVAILABLE True)
+  set(MPIX_CONTINUATIONS_LOOP_COUNT 79)
+else()
+  message(STATUS "MPIx Continuations not detected")
+  set(MPIX_CONTINUATIONS_AVAILABLE False)
+  set(MPIX_CONTINUATIONS_LOOP_COUNT 63)
+endif()
+
```

Comment:
Can we move this to all the other configuration tests please? This can use the same mechanisms (`pika_add_config_test` etc.). I think this can be a compile-only test, see other comment.

Reply:
Let's call the result `PIKA_WITH_MPIX_CONTINUATIONS`?

Reply:
oh dear. I didn't see this comment either. Personally, I prefer to have all the mpi related flags/checks/vars inside the mpi module and not in some remote place - where did you have in mind for it to go?

Reply:
ook. I moved it all to cmake/tests

Reply:
Thanks. I understand your motivation for keeping in the MPI module, but given all other feature checks and CMake options are defined currently with the "global" options, I think it's easier to manage if this test is also added to where everything else is.